src/Microsoft.ML.StandardLearners/Standard/SdcaBinary.cs

```diff
@@ -1574,8 +1509,131 @@ private protected override void CheckLabel(RoleMappedData examples, out int weig
+    }
+
+    public sealed class SdcaCalibratedBinaryTrainer :
+        SdcaBinaryTrainerBase<CalibratedModelParametersBase<LinearBinaryModelParameters, PlattCalibrator>>
+    {
+        internal const string LoadNameValue = "SDCALR";
+        internal const string UserNameValue = "Fast Linear (SA-SDCA) Logistic Regression";
+
+        /// <summary>
+        /// Configuration to training logistic regression using SDCA.
+        /// </summary>
+        public sealed class Options : BinaryArgumentBase
+        {
+        }
+
+        internal SdcaCalibratedBinaryTrainer(IHostEnvironment env,
+            string labelColumn = DefaultColumnNames.Label,
+            string featureColumn = DefaultColumnNames.Features,
+            string weightColumn = null,
+            float? l2Const = null,
+            float? l1Threshold = null,
+            int? maxIterations = null)
+             : base(env, labelColumn, featureColumn, weightColumn, new LogLoss(), l2Const, l1Threshold, maxIterations)
+        {
+        }
+
+        internal SdcaCalibratedBinaryTrainer(IHostEnvironment env, Options options)
+            : base(env, options, new LogLoss())
+        {
+        }
+
+        protected override CalibratedModelParametersBase<LinearBinaryModelParameters, PlattCalibrator> CreatePredictor(VBuffer<float>[] weights, float[] bias)
+        {
+            var linearModel = CreateLinearBinaryModelParameters(weights, bias);
+            var calibrator = new PlattCalibrator(Host, -1, 0);
+            return new ParameterMixingCalibratedModelParameters<LinearBinaryModelParameters, PlattCalibrator>(Host, linearModel, calibrator);
+        }
+
+        protected override SchemaShape.Column[] ComputeSdcaBinaryClassifierSchemaShape()
+        {
+            return new SchemaShape.Column[]
+            {
+                    new SchemaShape.Column(
+                        DefaultColumnNames.Score,
+                        SchemaShape.Column.VectorKind.Scalar,
+                        NumberType.R4,
+                        false,
+                        new SchemaShape(MetadataUtils.GetTrainerOutputMetadata())),
+                    new SchemaShape.Column(
+                        DefaultColumnNames.Probability,
+                        SchemaShape.Column.VectorKind.Scalar,
+                        NumberType.R4,
+                        false,
+                        new SchemaShape(MetadataUtils.GetTrainerOutputMetadata(true))),
+                    new SchemaShape.Column(
+                        DefaultColumnNames.PredictedLabel,
+                        SchemaShape.Column.VectorKind.Scalar,
+                        BoolType.Instance,
+                        false,
+                        new SchemaShape(MetadataUtils.GetTrainerOutputMetadata()))
+
+            };
+        }
+    }
+
+    public sealed class SdcaBinaryTrainer : SdcaBinaryTrainerBase<LinearBinaryModelParameters>
+    {
+        internal const string LoadNameValue = "SDCA";
```

Comment:
>SDCA [](start = 47, length = 4)

for backwards compatibility, i think this should have a new name, and the other one should be just called SDCA. I understand that you want to make the distinction, and the name is the right place to do so, but all the RSP using SDCA currently will fail if upgrading to this veriosn of TLC/ML.NET...

Reply:
After iteration 8, we have 100% BC for command line tools.

---
In reply to: [256265359](https://github.com/dotnet/machinelearning/pull/2506#discussion_r256265359) [](ancestors = 256265359)